apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/epoch_controller.ex

```diff
@@ -0,0 +1,87 @@
+defmodule BlockScoutWeb.API.RPC.EpochController do
+  use BlockScoutWeb, :controller
+
+  alias Explorer.{Chain, GenericPagingOptions}
+  alias Explorer.Chain.CeloElectionRewards
+
+  @api_voter_rewards_max_page_size 100
+
+  def getvoterrewards(conn, params) do
+    with {:voter_address_param, {:ok, voter_address_param}} <- fetch_address(params, "voterAddress"),
+         {:group_address_param, group_address_param} <- get_address(params, "groupAddress"),
+         {:voter_format, {:ok, voter_hash_list}} <- to_address_hash_list(voter_address_param, :voter_format),
+         {:group_format, {:ok, group_hash_list}} <- to_address_hash_list(group_address_param, :group_format),
+         {:block_number_param, {:ok, from}} <- fetch_block_number(params["from"], :up),
+         {:block_number_param, {:ok, to}} <- fetch_block_number(params["to"], :down),
+         %{page_size: page_size, page_number: page_number} <-
+           GenericPagingOptions.extract_paging_options_from_params(params, @api_voter_rewards_max_page_size),
+         rewards <-
+           CeloElectionRewards.get_epoch_rewards(voter_hash_list, group_hash_list, from, to, page_number, page_size) do
+      render(conn, :getvoterrewards, rewards: rewards)
+    else
+      {:voter_address_param, :error} ->
+        render(conn, :error, error: "Query parameter 'voterAddress' is required")
+
+      {:voter_format, :error} ->
+        render(conn, :error, error: "One or more voter addresses are invalid")
+
+      {:group_format, :error} ->
+        render(conn, :error, error: "One or more group addresses are invalid")
+
+      {:block_number_param, :error} ->
+        render(conn, :error, error: "Wrong format for block number provided")
+    end
+  end
+
+  defp get_address(params, key) when key == "groupAddress" do
```

Comment:
you can pattern match directly on these, so 

```elixir
defp get_address(params, key = "groupAddress"), do: {:group_address_param, Map.get(params, key)}
```

additionally it looks like the function for `group_address_param` has been duplicated here and at line 44

Reply:
actually I see that this is `get_address` and the other is `fetch_address` but both have the same logic - what's the reason for this?

Reply:
it's because `Map.fetch` returns either `{:ok, value}` or `{:error}` which is later used for error pattern matching (`voterAddress` is a required param) while `Map.get` will return `nil` in case the param is not provided and we process it anyway later on because it's not required.